include/tdi/common/tdi_notifications.hpp

```diff
@@ -17,72 +17,83 @@
  */
-class NotificationOutParams {
-public:
-    virtual ~NotificationOutParams() = default;
-    NotificationOutParams(const Table* table, const tdi_id_t &tdi_id_type)
-        : table_(table), notification_id_(tdi_id_type){};
+class NotificationParams {
+ public:
+  virtual ~NotificationParams() = default;
+
+  NotificationParams(const Table *table, tdi_id_t notification_id)
+      : table_(table), notification_id_(notification_id){};
+
+  virtual tdi_status_t setValue(const tdi_id_t & /*field_id*/,
+                                const uint64_t & /*value*/) {
+    return TDI_NOT_SUPPORTED;
+  }
 
-    virtual tdi_status_t setValue(const tdi_id_t & /*field_id*/,
-                                  const uint64_t & /*value*/) const {
-      return TDI_NOT_SUPPORTED;
-    };
-    virtual tdi_status_t getValue(const tdi_id_t & /*field_id*/,
-	                              uint64_t * /*value*/)  const {
-      return TDI_NOT_SUPPORTED;
-    };
+  virtual tdi_status_t setValue(const tdi_id_t & /*field_id*/,
+                                const std::vector<uint64_t> & /*value*/) {
+    return TDI_NOT_SUPPORTED;
+  }
 
-    virtual tdi_status_t setValue(const tdi_id_t & /*field_id*/,
-                                  const std::vector<uint8_t> & /*value*/) const {
-      return TDI_NOT_SUPPORTED; 
-	};
-    virtual tdi_status_t getValue(const tdi_id_t & /*field_id*/,
-                                  std::vector<uint8_t> * /*value*/) const {
-      return TDI_NOT_SUPPORTED;
-	};
+  virtual tdi_status_t getValue(const tdi_id_t & /*field_id*/,
+                                uint64_t * /*value*/) const {
+    return TDI_NOT_SUPPORTED;
+  }
 
-    virtual tdi_status_t getValue(const tdi_id_t & /*field_id*/,
-                                  uint8_t*const* /*value*/,
-                                  size_t * /*size*/) const {
-      return TDI_NOT_SUPPORTED;
-    };
+  virtual tdi_status_t getValue(const tdi_id_t & /*field_id*/,
+                                std::vector<uint64_t> * /*value*/) const {
+    return TDI_NOT_SUPPORTED;
+  }
+
+  const tdi_id_t &notificationIdGet() const { return notification_id_; };
+  const Table *table_;
 
  private:
-    const Table* table_; 
-    tdi_id_t notification_id_;
+  tdi_id_t notification_id_{0};
 };
 
 /**
- * @brief Contains TDI Notification callback function 
+ * @brief TDI Notification cpp callback function
  */
 typedef std::function<void(std::unique_ptr<tdi::TableKey> key,
                            std::unique_ptr<tdi::TableData> data,
-                           std::unique_ptr<tdi::NotificationOutParams> params,
-                           void *cookie)> tdiNotificationCallback;
-}  // tdi
+                           std::unique_ptr<tdi::NotificationParams> params,
+                           void *cookie)>
+    tdiNotificationCallback;
+}  // namespace tdi
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @brief TDI Notification c callback function
```

Comment:
Should it be a separate C frontend file ?

Reply:
c callback signature is needed by notificationRegisterC in tdi_table.hpp. 
  virtual tdi_status_t notificationRegisterC(
      const tdi::Target &target,
      const tdi_id_t &notification_id,
      const tdi_notification_callback &callback_fn,
      const tdi::NotificationParams &registration_params,
      void *cookie) const;